app/controllers/articles_controller.rb

```diff
@@ -0,0 +1,68 @@
+class ArticlesController < ApplicationController
+  load_and_authorize_resource
+  before_action :set_article, only: %i[ show edit update destroy ]
+
+  def index
+    @articles = Article.all
+  end
+
+  def show
+    @articles = Article.find(params[:id])
+  end
+
+  def new
+    @article = Article.new
+  end
+
+  def create
+    @article = Article.new(article_params)
+
+    # El metodo current_user, lo provee device a todos los controladores y vistas
+    # para acceder a el usuario logueado
+    @article.owner = current_user
+    respond_to do |format|
+      if @article.save
+        format.html { redirect_to @article, notice: "Article created." }
+        format.json { render :show, status: :created, location: @article }
+      else
+        format.html { render :new, status: :unprocessable_entity }
+        format.json { render json: @article.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  def edit
+    @article = Article.find(params[:id])
+  end
+
+  def update
+    respond_to do |format|
+      if @article.update(article_params)
+        format.html { redirect_to @article, notice: "Article updated." }
+        format.json { render :show, status: :ok, location: @article }
+      else
+        format.html { render :edit, status: :unprocessable_entity }
+        format.json { render json: @article.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  def destroy
+    @article.destroy
+    respond_to do |format|
+      format.html { redirect_to articles_url, notice: "Article deleted." }
+      format.json { head :no_content }
+    end
+  end
+
+  private
+  
+    def set_article
+      @article = Article.find(params[:id])
+    end
+
+
+    def article_params
+      params.require(:article).permit(:title, :body, :status)
```

Comment:
@sansuaza what the **require** and **permit** do? and why are they important?

Reply:
its functionality is to indicate what attributes the article class allows. Are needed to ensure that all atributes are given, in this case to update and create an article

Reply:
@sansuaza actually, this acts as params filtering --> only permit: title body and status from the article hash.